### PR TITLE
Editor: Support PluginPostStatusInfo Slot in the site editor

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632 
Similar to #59985

## What?

This PR adds support for the `PluginPostStatusInfo ` to the site editors allowing plugin authors to use the same code to load their plugins regardless of the context of the editor.

## How?

- Moves the PluginPostStatusInfo from `@wordpress/edit-post` to. `@wordpress/editor`

## Testing Instructions

You should be able to paste something like the following in both post and site editors (for pages) and have the button show up in the more menu.

```js
var __ = wp.i18n.__;
var PluginPostStatusInfo = wp.editor.PluginPostStatusInfo;

function MyPluginPostStatusInfo() {
	return React.createElement(
		PluginPostStatusInfo,
		{
			className: 'my-plugin-post-status-info',
		},
		__( 'My post status info' )
	);
}

wp.plugins.registerPlugin( 'toto', { render: () => wp.element.createElement( MyPluginPostStatusInfo ) } );
```

### CODE REVIEW

**Comment by ntsekouras** [comment]
The description is wrong 😄  - PluginMoreMenuItem -> PluginPostStatusInfo.

Do you think we rename the slot to something more close to `PluginPostSummaryInfo`? I was looking at this the other day and since we have `PostStatus` that actually handles the post's status attribute, this seemed just not the proper name. Same goes for `edit-post's` `PostStatus` component, but let's leave this out for now.

**Comment by youknowriad** [comment]
> Do you think we rename the slot to something more close to PluginPostSummaryInfo?

I think it could make sense but If I remember properly we used to consider the whole "summary" as "post status", not the post status property but the status of the post :P 

My gut feeling say it's not worth it, folks are already using that name and the change is seemless if it's just the package but I also don't feel that strongly.

File: docs/reference-guides/slotfills/README.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ const PostStatus = ( { isOpened, onTogglePanel } ) => (
 
 ## Currently available SlotFills and examples
 
-The following SlotFills are available in the `edit-post` package. Please refer to the individual items below for usage and example details:
+The following SlotFills are available in the `edit-post` or `editor` packages. Please refer to the individual items below for usage and example details:
```
Comment (aaronrobertshaw) [inline docs/reference-guides/slotfills/README.md:96]: Is there any value in splitting the list below based on package?

File: packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js [hunk 2]
```diff
@@ -24,18 +25,25 @@ export default function PageSummary( {
...
-				date={ date }
-				password={ password }
-				postId={ postId }
-				postType={ postType }
-			/>
-			<PostSchedulePanel />
-			<PostTemplatePanel />
-			<PostURLPanel />
-			<PostAuthorPanel />
+			<PluginPostStatusInfo.Slot>
```
Comment (ntsekouras) [inline packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js:28]: Is there a specific reason for wrapping these components with the slot? Is it about matching post editor and the concept of grouping this post information together or something else?

**Review by ntsekouras — APPROVED** [review]
Looks good, thanks!

File: packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js (hunk 2 above)
Comment (youknowriad) [reply packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js:28]: Good question, I did it to match the post editor. I'm guessing there's a small impact on the markup if I don't do it but I'm not sure if it's important.

**Review by aaronrobertshaw — APPROVED** [review]
Tests as advertised for me across both editors.

| Post Editor | Site Editor |
|---|---|
| Screenshot 2024-04-18 at 5 25 33 PM | Screenshot 2024-04-18 at 5 25 39 PM |

I think the slight change in markup isn't a show stopper and any renaming could be a future follow-up if it proves to be desired.

LGTM 🚢

File: docs/reference-guides/slotfills/README.md (hunk 1 above)
Comment (youknowriad) [reply docs/reference-guides/slotfills/README.md:96]: I think not, since they will ultimately be on the same package.

**Pull request merged**